source/exe/main.cc

```diff
@@ -33,10 +35,32 @@ class ProdComponentFactory : public ComponentFactory {
 
 } // Server
 
+int validateConfigAndExit(OptionsImpl& options, Server::ProdComponentFactory& component_factory,
+                          const LocalInfo::LocalInfoImpl& local_info) {
+  Server::ValidationHotRestart restarter;
```

Comment:
Do you need this restarter only for its `logLock()` and `accessLogLock()`? Could these be just independent log objects you create here instead?